ios-xcframework/podspec.erb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+Pod::Spec.new do |s|
+  s.name = 'Gutenberg'
+  s.summary = 'Printing since 1440'
+  s.homepage = 'https://github.com/wordpress-mobile/gutenberg-mobile'
+  s.authors = 'Automattic'
+
+  s.version = '<%= JSON.parse(File.read(File.join(File.dirname(File.dirname(File.expand_path(__dir__))), 'package.json')))['version'] %>'
```

Comment:
I find this file path a bit confusing, because the `__dir__` here is actually the directory of the `Fastfile`, not directory of this erb file. We can tell the ERB library the template file path, which would make this part of code easier to understand.

```diff
diff --git a/ios-xcframework/fastlane/Fastfile b/ios-xcframework/fastlane/Fastfile
index 913612dc..1fbd8a73 100644
--- a/ios-xcframework/fastlane/Fastfile
+++ b/ios-xcframework/fastlane/Fastfile
@@ -39,7 +39,9 @@ end
 def upload_podspec_to_s3(id:)
   context = binding

-  podspec_erb = ERB.new(File.read(File.join(XCFRAMEWORK_PROJECT_ROOT_FOLDER, 'podspec.erb')))
+  filepath = File.join(XCFRAMEWORK_PROJECT_ROOT_FOLDER, 'podspec.erb')
+  podspec_erb = ERB.new(File.read(filepath))
+  podspec_erb.filename = filepath

   Dir.mktmpdir do |tmp_dir|
     podspec_path = File.join(tmp_dir, 'podspec')
diff --git a/ios-xcframework/podspec.erb b/ios-xcframework/podspec.erb
index feecc3d4..525baa49 100644
--- a/ios-xcframework/podspec.erb
+++ b/ios-xcframework/podspec.erb
@@ -6,7 +6,7 @@ Pod::Spec.new do |s|
   s.homepage = 'https://github.com/wordpress-mobile/gutenberg-mobile'
   s.authors = 'Automattic'

-  s.version = '<%= JSON.parse(File.read(File.join(File.dirname(File.dirname(File.expand_path(__dir__))), 'package.json')))['version'] %>'
+  s.version = '<%= JSON.parse(File.read(File.expand_path('../package.json', __dir__)))['version'] %>'

   <%# This ought to be read from package.json, but there isn't one at the moment. %>
   s.license = { type: 'GPL2' }
```

Reply:
Great catch. I didn't realize the inconsistency. I kept tweaking the path till it worked 😅 

What do you think of this approach?

```diff
--- a/ios-xcframework/fastlane/Fastfile
+++ b/ios-xcframework/fastlane/Fastfile

@@ -39,12 +39,18 @@ end
 def upload_podspec_to_s3(id:)
   context = binding
 
-  podspec_erb = ERB.new(File.read(File.join(XCFRAMEWORK_PROJECT_ROOT_FOLDER, 'podspec.erb')))
-
-  Dir.mktmpdir do |tmp_dir|
-    podspec_path = File.join(tmp_dir, 'podspec')
-    File.write(podspec_path, podspec_erb.result(context))
-    upload_to_a8c_s3(file: podspec_path, name_on_s3: "Gutenberg-#{id}.podspec")
+  # chdir into the XCFramework project's root so that when the ERB access its
+  # path, it's the correct one relative to its location, not the one altered by
+  # the Fastlane execution.
+  Dir.chdir(XCFRAMEWORK_PROJECT_ROOT_FOLDER) do
+    podspec_erb = ERB.new(File.read(File.join(XCFRAMEWORK_PROJECT_ROOT_FOLDER, 'podspec.erb')))
+
+    Dir.mktmpdir do |tmp_dir|
+      podspec_path = File.join(tmp_dir, 'podspec')
+      File.write(podspec_path, podspec_erb.result(context))
+      upload_to_a8c_s3(file: podspec_path, name_on_s3: "Gutenberg-#{id}.podspec")
+    end
   end
 end
 
--- a/ios-xcframework/podspec.erb
+++ b/ios-xcframework/podspec.erb
@@ -6,7 +6,7 @@ Pod::Spec.new do |s|
   s.homepage = 'https://github.com/wordpress-mobile/gutenberg-mobile'
   s.authors = 'Automattic'
 
-  s.version = '<%= JSON.parse(File.read(File.join(File.dirname(File.dirname(File.expand_path(__dir__))), 'package.json')))['version'] %>'
+  s.version = '<%= JSON.parse(File.read(File.join(File.dirname(File.dirname(File.expand_path(__FILE__))), 'package.json')))['version'] %>'
 ```

Alternatively, I could take a step back on making the ERB smart, and make it expect the version value to be injected. Doing so would be self-explanatory, whereas both solutions we brought forward so far need some additional explanation/understating.

Reply:
I think reverting the ERB file to expect the version value makes sense to me. The template would become slightly simpler and all the variable setting are done in the Fastfile. 👍

Reply:
Addressed in a4570b03, plus realized that the iOS and Swift versions were already isolated, so read them from `xcconfig` in 70bf0607.